keras/optimizers/optimizer_experimental/adamw.py

```diff
@@ -173,7 +174,7 @@ def update_step(self, gradient, variable):
     alpha = (lr * tf.sqrt(1 - beta_2_power) / (1 - beta_1_power))
 
     # Apply step weight decay
-    if self.weight_decay != 0:
+    if self.weight_decay and variable not in self._exclude_from_weight_decay:
```

Comment:
self.weight_decay can not be used as a pythonic bool. This raises the error 

> OperatorNotAllowedInGraphError: Using a symbolic `tf.Tensor` as a Python `bool` is not allowed: AutoGraph did convert this function. This might indicate you are trying to use an unsupported feature

`self.weight_decay != 0` could be a solution.

Reply:
Also, please add a correctness test -- this sort of issue should be caught in unit tests.